nemo-python/src/lib.rs

```diff
@@ -104,7 +104,8 @@ struct NemoEngine(nemo::execution::DefaultExecutionEngine);
 impl NemoEngine {
     #[new]
     fn py_new(program: NemoProgram) -> PyResult<Self> {
-        let engine = ExecutionEngine::initialize(program.0).py_res()?;
+        let engine =
+            ExecutionEngine::initialize(program.0, ResourceProviders::from(vec![])).py_res()?;
```

Comment:
This should probably be `ResourceProviders::default`, or be controlled by the caller (maybe an optional argument). Otherwise the test cases (integration tests) will fail.